File: leetcode/go/3-longest-substring-without-repeating-characters.go
func lengthOfLongestSubstring(s string) int {
    j, start, end, max := 0, 0, 0, 0

    for i := 0; i < len(s); i++ {
        repeat := false
        for j = start; j < end && repeat == false; j++ {
            repeat = (s[i] == s[j])
        }

        if repeat {
            start = j
        }

        end++

        if end - start > max {
            max = end - start
        }
    }

    return max
}

File: leetcode/python/3-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s):
        mapping = {}
        result = 0
        start = 0
        
        for char in range(len(s)):
            if s[char] in mapping:
                start = max(mapping[s[char]] + 1,start)
                
            mapping[s[char]] = char
            result = max(result, char - start + 1)
                
        return result

print(Solution.lengthOfLongestSubstring(Solution, "123123412345"))